Sources/NIO/SocketChannel.swift

```diff
@@ -436,6 +436,19 @@ final class ServerSocketChannel: BaseSocketChannel<ServerSocket> {
             ch.close(promise: nil)
         }
     }
+
+    override func bufferPendingWrite(data: NIOAny, promise: EventLoopPromise<Void>?) {
+        promise?.fail(error: ChannelError.operationUnsupported)
+    }
+
+    override func markFlushPoint(promise: EventLoopPromise<Void>?) {
+        promise?.fail(error: ChannelError.operationUnsupported)
+    }
+
+    override func writeToSocket() throws -> OverallWriteResult {
+        // We just return .writtenCompletely as there is never anything to write and this is the cheapest thing to do here.
+        return .writtenCompletely
```

Comment:
Does this function ever get called? I think we can `preconditionFailure` here on the grounds that we should never enter this.

Reply:
@Lukasa it actually was from `flush0()`. That said I just override `flush0()` now for `ServerSocketChannel`.